python/mcap/mcap0/stream_reader.py

```diff
@@ -96,6 +102,37 @@ def records(self) -> Iterator[McapRecord]:
                 self.__footer = record
                 read_magic(self.__stream)
 
+    @property
+    def decoded_records(self) -> Iterator[Any]:  # type: ignore
+        """
+        Tries to decode records encountered in the MCAP in order. May throw an error
+        if unable to decode the data.
+        """
+        channels: Dict[int, Channel] = {}
+        schemas: Dict[int, Schema] = {}
+        msg_types: Dict[str, Any] = {}
+        for record in self.records:
+            if isinstance(record, Schema):
+                schemas[record.id] = record
+                if record.encoding != "ros1":
+                    raise McapError(
+                        f"Can't decode schema with encoding {record.encoding}"
+                    )
+                if dynamic is None:
+                    raise McapError("ros1 libraries not found")
+                if not record.name in msg_types:
+                    msg_type = dynamic.generate_dynamic(  # type: ignore
+                        record.name, record.data.decode()
+                    )
+                    msg_types[record.name] = msg_type[record.name]
+            if isinstance(record, Channel):
+                channels[record.id] = record
+            if isinstance(record, Message):
+                channel = channels[record.channel_id]
+                schema = schemas[channel.schema_id]
+                message = msg_types[schema.name]().deserialize(record.data)
```

Comment:
What happens if this fails?

Reply:
Should probably wrap this in a try/catch that throws an `McapError` right?